crates/subspace-archiving/src/archiver.rs

```diff
@@ -725,6 +719,31 @@ impl Archiver {
             pieces
         };
 
+        let existing_commitments = self.incremental_record_commitments.len();
+        // Add commitments for pieces that were not created incrementally
+        {
+            #[cfg(not(feature = "rayon"))]
+            let source_pieces = pieces.source();
+            #[cfg(feature = "rayon")]
+            let source_pieces = pieces.par_source();
+
+            let iter = source_pieces
+                .skip(existing_commitments)
+                .map(|piece| piece.record().safe_scalar_arrays().map(Scalar::from))
+                .map(|scalars| {
+                    let polynomial = self.kzg.poly(&scalars.collect::<Vec<_>>()).expect(
+                        "KZG instance must be configured to support this many scalars; qed",
+                    );
+                    self.kzg
+                        .commit(&polynomial)
+                        .expect("KZG instance must be configured to support this many scalars; qed")
+                });
+
+            #[cfg(not(feature = "rayon"))]
+            iter.collect_into(&mut *self.incremental_record_commitments);
+            #[cfg(feature = "rayon")]
+            iter.collect_into_vec(&mut *self.incremental_record_commitments);
```

Comment:
`.collect_into()` is unstable, hopefully once stabilized `rayon` will adopt the same name too.